src/runtime_src/core/edge/skd/sk_daemon.cpp

```diff
@@ -149,13 +149,17 @@ static void softKernelLoop(char *name, char *path, uint32_t cu_idx)
   }
 
   syslog(LOG_INFO, "%s_%d start running\n", name, cu_idx);
-    
+
   /* Set Kernel Ops */
   ops.getHostBO = &getHostBO;
   ops.mapBO     = &mapBO;
   ops.freeBO    = &freeBO;
 
   args_from_host = (unsigned *)getKernelArg(boh, cu_idx);
+  if (args_from_host == MAP_FAILED) {
+      syslog(LOG_ERR, "Failed to map soft kernel args for %s_%d", name, cu_idx);
+      return;
```

Comment:
You may want to free register BO and close sk_handle before bail out.

Reply:
@larry9523 
I've updated the pull request with the error path handling.

Reply:
@rathaval I think it is better to keep the original code but just close and free before return. The reason for that is in destroySoftKernel()l, we will do munmap. Since here we are failing because we can not do mmap, it should not do munmap in this situation.

Reply:
Done. New patchset uploaded.